c/src/ml-lxm-service.c

```diff
@@ -0,0 +1,276 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/**
+ * @file    ml-lxm-service.c
+ * @date    23 JULY 2025
+ * @brief   Machine Learning LXM(LLM, LVM, etc.) Service API
+ * @see     https://github.com/nnstreamer/api
+ * @author  Hyunil Park <hyunil46.park@samsung.com>
+ * @bug     No known bugs except for NYI items
+ */
+#include <glib.h>
+#include <string.h>
+#include <stdlib.h>
+#include <ml-api-service.h>
+#include "ml-lxm-service-internal.h"
+
+/**
+ * @brief Internal structure for the session.
+ */
+typedef struct
+{
+  ml_service_h service_handle;
+  char *config_path;
+  char *instructions;
+} ml_lxm_session_internal;
+
+/**
+ * @brief Internal structure for the prompt.
+ */
+typedef struct
+{
+  GString *text;
+} ml_lxm_prompt_internal;
+
+/**
+ * @brief Checks LXM service availability.
+ * @param[out] status Current availability status.
+ * @return ML_ERROR_NONE on success, error code otherwise.
+ */
+int
+ml_lxm_check_availability (ml_lxm_availability_e * status)
+{
+  *status = ML_LXM_AVAILABILITY_AVAILABLE;
+
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Creates an LXM session.
+ * @param[out] session Session handle.
+ * @param[in] config_path Path to configuration file.
+ * @param[in] instructions Initial instructions (optional).
+ * @return ML_ERROR_NONE on success.
+ */
+int
+ml_lxm_session_create (ml_lxm_session_h * session, const char *config_path,
+    const char *instructions)
+{
+  ml_lxm_session_internal *s;
+  int ret;
+
+  if (!session || !config_path)
+    return ML_ERROR_INVALID_PARAMETER;
+
+  s = g_malloc0 (sizeof (ml_lxm_session_internal));
+  if (!s)
+    return ML_ERROR_OUT_OF_MEMORY;
+
+  s->config_path = g_strdup (config_path);
+  s->instructions = instructions ? g_strdup (instructions) : NULL;
+
+  ret = ml_service_new (s->config_path, &s->service_handle);
```

Comment:
Suggestion: create service handle before allocating memory.
```
ml_service_h handle;

ret = ml_service_new (config_path, &handle);
if (ret != ML_ERROR_NONE) {
  return ret;
}

*session = s = g_new0 (ml_lxm_session_internal, 1);
if (!s) {
  ml_service_destroy (handle);
  return ML_ERROR_OUT_OF_MEMORY;
}

s->config_path = g_strdup (config_path);
s->instructions = g_strdup (instructions);
s->service_handle = handle;

```

Reply:
Okay, Good. Thank you.